test/e2e/pivoting_test.go

```diff
@@ -281,3 +290,108 @@ func labelBMOCRDs(targetCluster framework.ClusterProxy) {
 		Expect(err).To(BeNil(), "Cannot label BMO CRDs")
 	}
 }
+
+func rePivoting() {
+
+	By("By removing Ironic deployment from target cluster")
+	removeIronicDeploymentOnTarget()
+
+	By("Reinstate Ironic containers and BMHD")
+	ephemeralCluster := os.Getenv("EPHEMERAL_CLUSTER")
+	if ephemeralCluster == "kind" {
+		bmoPath := os.Getenv("BMOPATH")
+		ironicCommand := bmoPath + "/tools/run_local_ironic.sh"
+		cmd := exec.Command("sh", "-c", "export CONTAINER_RUNTIME=docker; "+ironicCommand)
+		stdoutStderr, err := cmd.CombinedOutput()
+		fmt.Printf("%s\n", stdoutStderr)
+		Expect(err).To(BeNil(), "Cannot run local ironic")
+	} else {
+
+		LogFromFile(filepath.Join(artifactFolder, "clusters", clusterName+"-pivoting", "clusterctl-init.log"))
+
+		By("Configure Ironic Configmap")
+		configureIronicConfigmap(true)
+
+		By("Install Ironic in the target cluster")
+		installIronicBMO(bootstrapClusterProxy, "true", "false")
+
+		By("Reinstate Ironic Configmap")
+		configureIronicConfigmap(false)
+	}
+
+	By("Ensure API servers are stable before doing move")
+	// Nb. This check was introduced to prevent doing move to self-hosted in an aggressive way and thus avoid flakes.
+	// More specifically, we were observing the test failing to get objects from the API server during move, so we
+	// are now testing the API servers are stable before starting move.
+	Consistently(func() error {
+		kubeSystem := &corev1.Namespace{}
+		return bootstrapClusterProxy.GetClient().Get(ctx, client.ObjectKey{Name: "kube-system"}, kubeSystem)
+	}, "5s", "100ms").Should(BeNil(), "Failed to assert bootstrap API server stability")
+
+	By("Moving back to bootstrap cluster")
+	clusterctl.Move(ctx, clusterctl.MoveInput{
+		LogFolder:            filepath.Join(artifactFolder, "clusters", clusterName+"-bootstrap"),
+		ClusterctlConfigPath: clusterctlConfigPath,
+		FromKubeconfigPath:   targetCluster.GetKubeconfigPath(),
+		ToKubeconfigPath:     bootstrapClusterProxy.GetKubeconfigPath(),
+		Namespace:            namespace,
+	})
+
+	LogFromFile(filepath.Join(artifactFolder, "clusters", clusterName+"-bootstrap", "clusterctl-move.log"))
+
+	By("Checking that the repivoted cluster is up and running")
+	pivotingCluster := framework.DiscoveryAndWaitForCluster(ctx, framework.DiscoveryAndWaitForClusterInput{
+		Getter:    bootstrapClusterProxy.GetClient(),
+		Namespace: namespace,
+		Name:      clusterName,
+	}, e2eConfig.GetIntervals(specName, "wait-cluster")...)
+
```

Comment:
Should we have an Expect assertion here to check if pivotingCluster is not nil ?

Reply:
I don't think it is needed, this DiscoveryAndWaitForCluster is full of checks and it can do waiting by itself too so I don't think we need a wrapper.